nifi-nar-bundles/nifi-site-to-site-reporting-bundle/nifi-site-to-site-reporting-task/src/main/java/org/apache/nifi/reporting/SiteToSiteProvenanceReportingTask.java

```diff
@@ -263,10 +271,15 @@ static JsonObject serialize(final JsonBuilderFactory factory, final JsonObjectBu
 
         addField(builder, "actorHostname", hostname);
         if (nifiUrl != null) {
-            final String urlPrefix = nifiUrl.toString().replace(nifiUrl.getPath(), "");
-            final String contentUriBase = urlPrefix + "/nifi-api/controller/provenance/events/" + event.getEventId() + "/content/";
-            addField(builder, "contentURI", contentUriBase + "output");
-            addField(builder, "previousContentURI", contentUriBase + "input");
+            // TO get URL Prefix, we just remove the /nifi from the end of the URL. We know that the URL ends with
+            // "/nifi" because the Property Validator enforces it
+            final String urlString = nifiUrl.toString();
+            final String urlPrefix = urlString.substring(0, urlString.length() - 5);
```

Comment:
Understood that validator enforces "/nifi" yet still seems to me like a place to use the URL methods to build URL or perhaps reference a static variable representing that string here (and in the Validator). Low likelihood for that path to change but if it happened this would at least be a quick fix or none required at all.